homeassistant/components/concord232/alarm_control_panel.py

```diff
@@ -89,32 +93,38 @@ def __init__(self, url, name, code, mode):
         self._alarm = concord232_client.Client(self._url)
         self._alarm.partitions = self._alarm.list_partitions()
 
-    def update(self) -> None:
-        """Update values from API."""
+    @property
+    def alarm_status(self) -> str:
+        """Get current status from API."""
         try:
             part = self._alarm.list_partitions()[0]
         except requests.exceptions.ConnectionError as ex:
             _LOGGER.error(
                 "Unable to connect to %(host)s: %(reason)s",
                 {"host": self._url, "reason": ex},
             )
-            return
+            return STATE_UNAVAILABLE
         except IndexError:
             _LOGGER.error("Concord232 reports no partitions")
-            return
+            return STATE_UNAVAILABLE
 
         if part["arming_level"] == "Off":
-            self._attr_state = STATE_ALARM_DISARMED
-        elif "Home" in part["arming_level"]:
-            self._attr_state = STATE_ALARM_ARMED_HOME
-        else:
-            self._attr_state = STATE_ALARM_ARMED_AWAY
+            return STATE_ALARM_DISARMED
+        if "Home" in part["arming_level"]:
+            return STATE_ALARM_ARMED_HOME
+
+        return STATE_ALARM_ARMED_AWAY
+
+    def update(self) -> None:
+        """Update alarm panel state."""
+        self._attr_state = self.alarm_status
```

Comment:
This feels like a proxy method now. Why not do all of what you did in `alarm_status()` directly here?